bats/sanity-check/20-selftests.bats

```diff
@@ -1525,3 +1525,50 @@ selftest_cpuset_unsorted() {
         test_yaml_regexp "/tests/0/threads/$i/messages/0/text" "I> ${cpuset[$i]}\$"
     done
 }
+
+selftest_interrupt_common() {
+    local timeout=30000
+    local signal=$1
+    if $is_windows; then
+        skip "Unix-only test"
+    fi
+
+    # We can't use sandstone_selftest or run_sandstone_yaml here - must run directly
+    local yamlfile=`mktempfile output-XXXXXX.yaml`
+    trap "rm -f -- $yamlfile" EXIT
+    $SANDSTONE -Y -o - -n1 --max-test-loop-count=0 --selftests -e selftest_timedpass -t $timeout > $yamlfile &
+    local pid=$!
+
+    # let's wait for the test to actually start
+    sleep 1
+
+    kill -$signal $pid
+    if wait $!; then
+        # Confirm it was the correct signal
+        (( $? == (128 | $(kill -l $signal))))
+    else
+        echo >&2 'Test did not exit!'
+    fi
```

Comment:
It indeed looks wrong. We're probably running the `else` code all the time, but because this is passing, I didn't notice it. Weirdly, I can't debug this because if I add a `false` after this `if` so it will fail, bats fails completely:
```
1..3
# bats warning: Executed 0 instead of expected 3 tests
```

Changing to fd 3 does show that we're running on the wrong side.

Reply:
https://github.com/bats-core/bats-core/issues/364 - I need to remove the `trap`